lib/index.js

```diff
@@ -281,28 +386,40 @@ api.get = async ({actor, account, email, type} = {}) => {
  * @param actor the actor or capabilities to perform the action.
  * @param account the ID of the account to remove a token from.
  * @param type the type of token to remove.
+ * @param tokenId the id of the token to be removed.
  *
  * @return a Promise that resolves once the operation completes.
  */
-api.remove = async ({actor, account, type} = {}) => {
+api.remove = async ({actor, account, type, tokenId} = {}) => {
   assert.string(account, 'account');
+  assert.optionalString(tokenId, 'tokenId');
   validateTokenType(type);
 
   await brPermissionCheck(
     actor, PERMISSIONS.ACCOUNT_UPDATE, {resource: [account]});
-
   // remove token from meta
-  const result = await database.collections.account.updateOne({
-    id: database.hash(account)
-  }, {
-    $unset: {
-      [`meta.bedrock-authn-token.tokens.${type}`]: true
+  let result;
+  if(type === 'nonce') {
+    result = await database.collections.account.updateOne({
+      id: database.hash(account),
+    }, {
+      $pull: {
+        [`meta.bedrock-authn-token.tokens.${type}`]: {id: tokenId}
+      }
+    }, database.writeOptions);
+  } else {
+    result = await database.collections.account.updateOne({
+      id: database.hash(account)
+    }, {
+      $unset: {
+        [`meta.bedrock-authn-token.tokens.${type}`]: true
+      }
+    }, database.writeOptions);
```

Comment:
The only difference between the if/else here seems to be the second parameter to the `updateOne` function -- so could this be reworked to simplify to that? i.e., use the if/else to build up an `update` var only and then the common code calls `updateOne` with the other params and the `update` var.

Reply:
Updated.